File: Leetcode/Weekly Contest 241/WC241-B.cpp
class Solution {
public:
	int minSwaps(string s) {
		int n1 = 0, n2 = 0;
		int ans = 1e9;
		for (int i = 0; i < s.size(); i += 2) if (s[i] == '0') n1++;
		for (int i = 1; i < s.size(); i += 2) if (s[i] == '1') n2++;
		if (n1 == n2) ans = min(ans, n1);

		n1 = 0, n2 = 0;
		for (int i = 0; i < s.size(); i += 2) if (s[i] == '1') n1++;
		for (int i = 1; i < s.size(); i += 2) if (s[i] == '0') n2++;
		if (n1 == n2) ans = min(ans, n1);
		return ans == 1e9 ? -1 : ans;
	}
};

File: Leetcode/Weekly Contest 241/WC241-B.py
class Solution:
    def minSwaps(self, s: str) -> int:
        n1 = sum([1 if s[i] == '0' else 0 for i in range(0, len(s), 2)])
        n2 = sum([1 if s[i] == '1' else 0 for i in range(1, len(s), 2)])
        ans1 = n1 if (n1 == n2) else 1e9
        
        n1 = sum([1 if s[i] == '1' else 0 for i in range(0, len(s), 2)])
        n2 = sum([1 if s[i] == '0' else 0 for i in range(1, len(s), 2)])
        ans2 = n1 if (n1 == n2) else 1e9
        
        if ans1 == 1e9 and ans2 == 1e9:
            return -1
        return min(ans1, ans2)